dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/model/IntegrationResourceLinkTest.java

```diff
@@ -0,0 +1,202 @@
+        ContentType dotAssetContentType     = contentTypeAPI
+                .save(ContentTypeBuilder.builder(DotAssetContentType.class).folder(FolderAPI.SYSTEM_FOLDER)
+                        .host(host.getIdentifier()).name(variable)
+                        .owner(user.getUserId()).build());
+        final Map<String, Field> fieldMap = dotAssetContentType.fieldMap();
+        com.dotcms.contenttype.model.field.Field binaryField           = fieldMap.get(DotAssetContentType.ASSET_FIELD_VAR);
+        final FieldVariable allowFileTypes = ImmutableFieldVariable.builder().key(BinaryField.ALLOWED_FILE_TYPES)
+                .value(accept).fieldId(binaryField.id()).build();
+        binaryField.constructFieldVariables(Arrays.asList(allowFileTypes));
+
+        dotAssetContentType = contentTypeAPI.save(dotAssetContentType);
+        binaryField = fieldAPI.save(binaryField, user);
+        fieldAPI.save(allowFileTypes, user);
+
+        return Tuple.of(binaryField, dotAssetContentType);
+    }
+
+    /**
+     * Method to test:  build
+     * Given Scenario: Creates a dotAsset with a binary field as a txt and figured out the version path and id path, the user is an admin
+     * ExpectedResult: paths should have some tokens such as dA, shorty id or inode and the file name, download must be allowed
+     * @throws Exception
+     */
+    @Test
+    public void test_Text_ResourceLink_Expect_Downloadable_No_Port_Number_Dot_Asset() throws Exception{
+
+        final Host host = APILocator.systemHost();
+        final String mimeType = "text/plain";
+        final boolean isSecure = false;
+        final File file = FileUtil.createTemporalFile("comments-list", "txt", "This is a test temporal file");
+        final String htmlFileName = file.getName();
+        final User adminUser = TestUserUtils.getAdminUser();
+
+        final Tuple2<Field, ContentType> fieldDotTextAssetContentType = this.createDotAssetContentType(host,
+                "text/*", "textDotAsset" + System.currentTimeMillis());
+
+        final Contentlet contentlet = new ContentletDataGen(fieldDotTextAssetContentType._2().variable())
+                .setProperty(DotAssetContentType.ASSET_FIELD_VAR, file).nextPersisted();
+
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        when(request.getAttribute(ResourceLink.HOST_REQUEST_ATTRIBUTE)).thenReturn(host.getIdentifier());
+        when(request.isSecure()).thenReturn(isSecure);
+        when(request.getServerPort()).thenReturn(80);
+
+
+        final ResourceLinkBuilder resourceLinkBuilder = new ResourceLinkBuilder();
+        final ResourceLink link = resourceLinkBuilder.build(request, adminUser, contentlet);
+
+        assertEquals(mimeType, link.getMimeType());
+        assertFalse(link.isDownloadRestricted());
+
+        assertTrue(link.getIdPath().contains("/dA/") && link.getIdPath().contains("/"+htmlFileName)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "/dA/" appears 4 times in this file; the first occurrence is on line 114](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5372524)